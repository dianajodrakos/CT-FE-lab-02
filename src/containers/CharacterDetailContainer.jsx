import React from 'react';
import Villager from '../components/villagers/Villager.jsx';
import { useVillager } from '../hooks/useVillager';
import Loading from '../components/loading/Loading.jsx';

const CharacterDetailContainer = () => {
  const { loading, villager } = useVillager();

  if(loading)
    return <Loading />;
    
  return <Villager
    id={villager.id}
    name={villager.name}
    image={villager.image}
    quote={villager.quote}
    style={villager.style}
    coffee={villager.coffee}
  />;
};

export default CharacterDetailContainer;



