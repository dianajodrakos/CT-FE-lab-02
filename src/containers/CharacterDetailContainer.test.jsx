import React from 'react';
import { Router, Route } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import { createMemoryHistory } from 'history';
import CharacterDetailContainer from './CharacterDetailContainer.jsx';
import detailData from '../fixtures/detailData.json';

const server = setupServer(
  rest.get('https://ac-vill.herokuapp.com/villagers/:name', (req, res, ctx) => {
    return res(
      ctx.status(200),
      ctx.json(detailData)
    );
  })
);

function renderWithRouter(
  ui,
  {
    route = '/',
    path = '/',
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) {
  return {
    ...render(
      <Router history={history}>
        <Route path={path} component={ui} />
      </Router>
    )
  };
}

describe('Animal Crossing Character Detail Container', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());

  it('renders a single villager', async () => {
    renderWithRouter(CharacterDetailContainer, 
      {
        route: '/Agnes',
        path: '/:name'
      }
    );

    screen.getByAltText('loading');

    const h2 = await screen.findByRole('heading');

    expect(h2).toMatchSnapshot();
  });
});
