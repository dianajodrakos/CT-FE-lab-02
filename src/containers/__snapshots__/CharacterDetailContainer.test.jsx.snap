// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Animal Crossing Character Detail Container renders a single villager 1`] = `
<h2
  aria-label="heading"
>
  Agnes
</h2>
`;
